Update README.md (#4)
guardian/our-engineering-culture: README.md

**Delivery mindset**

We believe in being pragmatic and moving fast, spending our energy on the right things, shipping code and then rapidly iterating on it. 

_Behaviours_:
- We simplify our work, focusing on getting it to the user as early as possible.
- We practice [continuous delivery](https://en.wikipedia.org/wiki/Continuous_delivery)
- We are trying to [make the change easy, before the easy change](https://twitter.com/KentBeck/status/250733358307500032).   
- We aim to solve issues rapidly prioritising them based on impact they may have.

**Inclusiveness & Respect** 

We believe that our culture is important, and we all contribute to it regardless of our role or level.

_Behaviours_:
 - We ensure to treat everyone we encounter with respect independently of their skillsets and levels of ability
 - Every one of us has multiple interactions with a variety of people across the department and company every day. Each one of those is an opportunity to express and refine our culture. This written statement of our culture is shared, and anyone can [contribute](CONTRIBUTING.md).

>  - We respect the opinion of others and are gracious in our interactions, whether meeting physically or digitally, whether friend or stranger, whether employee, customer or commenter.

Reviewer: I would personally restrict respect to opinions about product and engineering and other generic topics etc,.  It doesn't come up often but presumably we want to be explicit about not protecting uninclusive opinions expressed.

Author: As you know `uninclusive opinions` could be a divisive topic because outside of what is covered by law `uninclusive` is subject to interpretation based on culture, time and everyone's background. The Guardian is an independent newspaper and media organisation and as such may choose to publish articles or not moderate comments that may be felt by some as `uninclusive`. As an employee one have the right to (strongly) disagree with some editorial choice, a moderation decision, a customer review, or with an other employee's opinion but that disagreement should be expressed in a way which is respectful to the other party.

>  - We come from different backgrounds, work in different roles, are different ages, and therefore have different levels of experience. All in all, we pull each other up, not bring each other down.

Reviewer: someone reading this might feel it's a bit "etc" on the underrepresented group(s) they identify with, if we care we can specify more groups we would welcome in this point

Author: I think the mention of `background`, `roles` and `ages` is linked here to support the different `levels of experience`. I am not sure mentioned `underrepresented group(s)` will make this sentence particularly better, however I think we could another sentence.

Reviewer 2: How about we rephrase this to take out the group characteristics and instead say "Our experiences and backgrounds differ"  to address John's point above?

Also I think we can call out more than an appreciation of the impact this has on level of experience here, how about:

"and therefore so do our perspectives, areas of expertise, and levels of experience"

So all together:

"Our experiences and backgrounds differ, and therefore so do our perspectives, areas of expertise, and levels of experience. All in all ..."

Author: @amyhughes Thanks, I like your wording and have now used it!

**Working as a team**

We believe that the team is everything and our lowest common unit, set up with the necessary skills to succeed.

_Behaviours_:
 - We fail and succeed together as a team.
 - We are independent, accountable and share responsibility. We conduct ourselves with humility, kindness and a calm manner. We debate ideas, not people and we share constructive feedback openly. 
 - We take the time needed to listen to and help our colleagues with any problem they might have from anywhere within the business.
 - We value the experience of people in the team and the different constraints within their lives. 
 - We celebrate success and positive behaviour, recognizing everyone’s contribution while treating every moment including tasks and failures as a learning opportunity.

> **We are data driven** 

Reviewer 3: Are we Data *Driven* or Data *Informed*?

One definition:

https://segment.com/resources/data-strategy/data-driven-vs-data-informed/

> Data-driven means that you always rely on data to make your decisions. If you’re trying to improve the UX of your product, you’ll run tests to get data to find the best approach.

> Data-informed means that you use data alongside your unique experience, user research, and other inputs to make decisions. With a data-informed approach, data is just one part of your decision-making process.

I believe that we're much more aligned to the Data Informed approach - we use the data to help make decisions alongside the historic, editorial, user experience and performance contexts.

Author: @gtrufitt Thanks Gareth, great point! I have now amended to use `Data informed`

 - We base our decisions on analytics and experiments to reach consensus having heard everyone’s voice. 
 - We measure our success through data analysis and not through the highest paid person's opinion or the loudest voice.

**Empowerment**

We believe that all of us want to - and can - make a difference to the Guardian.

-Example Behaviours_: 
  - Meeting the needs of people we encounter across the Guardian is consistently encouraged across all levels of engineering.
  - We trust engineers to build the right things, collaborating closely and transparently with other teams.
  - We own the problem, not the spec. We don’t wait for decisions to be made for us, we are an empowered engineering function and you are an empowered engineer. We use that power responsibly as we are conscious some decisions may need to be signed off and work collaboratively with our stakeholders.

**Technical Excellence** 

We believe that embracing technical innovation and excellence can help us make a difference.

_Behaviours_:
- We think we can better achieve the Guardian's objectives by having a deep understanding of the technology we use and adopting some emerging tools and solutions before others:

>     - We have been one of the [first company in UK to adopt `scala`](https://www.theguardian.com/info/developer-blog/2013/apr/26/how-the-guardian-built-scalable-foundation)

Reviewer 4: ```suggestion
    - We were one of the [first companies in UK to adopt `scala`](https://www.theguardian.com/info/developer-blog/2013/apr/26/how-the-guardian-built-scalable-foundation)
```
Grammar fix

>     - We have been one of the first company [to use continuous delivery](https://www.theguardian.com/info/developer-blog/2015/jan/05/delivering-continuous-delivery-continuously)

Reviewer 4: ```suggestion
    - We were one of the first companies [to use continuous delivery](https://www.theguardian.com/info/developer-blog/2015/jan/05/delivering-continuous-delivery-continuously)
```

>     - We have been one of the [first legacy company to move entirely to the cloud (AWS)](http://perspectives.mvdirona.com/2016/03/a-decade-of-innovation/)

Reviewer 4: ```suggestion
    - We were one of the [first legacy companies to move entirely to the cloud (AWS)](http://perspectives.mvdirona.com/2016/03/a-decade-of-innovation/)
```

>     - We have been one of the pioneer of [the concept of microservices](https://martinfowler.com/articles/microservices.html#footnote-etymology)

Reviewer 4: ```suggestion
    - We were a pioneer of [the concept of microservices](https://martinfowler.com/articles/microservices.html#footnote-etymology)
```

Reviewer: was that footnote anchor the intended destination of the link?

 - We maintain our technical advantage through continuous innovation
 - We aim to be at the forefront of tech, staying curious, looking to innovate and continuously improve.

**Knowledge sharing**

We believe that it’s our responsibility to upskill one another, sharing our knowledge, learnings and mistakes. 

_Behaviours_:

>  - Individuals are entitled to pairing time for their development. 

Reviewer 4: It would be nice to emphasise that we also are dedicated to developing best practice for pairing. Good quality pairing that isn't passive or unequal in "burden" is really important, especially at a junior level

Reviewer 5: I'm not sure how best to word this, but could we say that we encourage pairing, rather than that we are entitled to it?

Author: @buck06191 could you suggest a sentence to describe your thinking?
@jennygrahamjones Would `Individuals are encouraged to pair, as part of their development` be fine, or do you have a better suggestion?

Author: @jennygrahamjones thanks, I have now updated.

Reviewer 4: @mchv Something along the lines of 
```
 - Individuals are encouraged to pair as part of their development and are empowered to take an active role in what this looks like and to work together to find their own best practice.
```

If someone has more concrete suggestions for best practice, that could be included, but I also think it's about empowering people to take charge of how they pair and what works best for them as I've found this varies depending on individuals, teams and the work being undertaken.

Reviewer 2: I think "encouraged" is a bit thin, "entitled" gave the sense that the department would ensure that pairing was available rather than it being up to the individual to try to forge this culture. The issue I have with "entitled" is that it makes pairing out to be something that is offered reluctantly.

I think we could do more to pull out the shared benefits of pairing here, currently it sound like we just see it as an aid to development. How about:

"We practice pairing to help share our knowledge, grow our skills, and facilitate collaborative decision making"

Reviewer 5: @amyhughes that captures much more eloquently what I was thinking of with 'encouraged' :)

Reviewer 2: Sorry Josh, I've just noticed your amendment. How about a second sentence which says something like:

"We value pairing as a skill which we encourage people to grow, working together to find the styles which are most effective."

@rtyley might have something to add here

Author: Thanks @amyhughes for the suggestions, I have now updated the wording.

 - We make time for individual-learning & career development with personal development being our priority.
 - We write code for the most junior, not the most senior.
 - We actively share at all levels of users and stakeholders, whether it's our knowledge, work, planning or approach.

**We care**
 

> We choose to work here because we believe in [its values](https://workforus.theguardian.com/our-values) and what the Guardian stands for. We are not driven to simply deliver the implementation of a specification, but delight in working collaboratively to achieve an objective. In return, we are enthused to do whatever we can to deliver value to the Guardian.

Reviewer 4: ```suggestion
We choose to work here because we believe in [our values](https://workforus.theguardian.com/our-values) and what the Guardian stands for. We are not driven to simply deliver the implementation of a specification, but delight in working collaboratively to achieve an objective. In return, we are enthused to do whatever we can to deliver value to the Guardian.
```

Makes the values more inclusive and matches the link location

- We are trying to [make the change easy, before the easy change](https://twitter.com/KentBeck/status/250733358307500032).   
- We aim to solve issues rapidly prioritising them based on impact they may have.

**Inclusiveness & Respect** 

We believe that our culture is important, and we all contribute to it regardless of our role or level.

_Behaviours_:
 - We ensure to treat everyone we encounter with respect independently of their skillsets and levels of ability
 - Every one of us has multiple interactions with a variety of people across the department and company every day. Each one of those is an opportunity to express and refine our culture. This written statement of our culture is shared, and anyone can [contribute](CONTRIBUTING.md).
 - We respect the opinion of others and are gracious in our interactions, whether meeting physically or digitally, whether friend or stranger, whether employee, customer or commenter.
 - We come from different backgrounds, work in different roles, are different ages, and therefore have different levels of experience. All in all, we pull each other up, not bring each other down.

>  - We value and respect all differences in all people (seen and unseen). We aspire to inclusive working experiences and an environment that reflects the audience we serve, where our people have equal access to career development opportunities, their voices are heard and can contribute to our future. We aim to hire, retain and advance people from underrepresented groups.

Author: @johnduffell what do you think of this suggestion, regarding your previous comment?

Reviewer: great.  if we are happy to strengthen it to "does happen" rather than "can" that would be even better
```suggestion
 - We value and respect all differences in all people (seen and unseen). We aspire to inclusive working experiences and an environment that reflects the audience we serve, where our people take up career development opportunities equally, their voices are heard and contribute to our future. We aim to hire, retain and advance people from underrepresented groups.
```

**Delivery mindset**

We believe in being pragmatic and moving fast, spending our energy on the right things, shipping code and then rapidly iterating on it. 

_Behaviours_:
- We simplify our work, focusing on getting it to the user as early as possible.
- We practice [continuous delivery](https://en.wikipedia.org/wiki/Continuous_delivery)
- We are trying to [make the change easy, before the easy change](https://twitter.com/KentBeck/status/250733358307500032).   

> - We aim to solve issues rapidly prioritising them based on impact they may have.

Reviewer 6: Suggest using "problems" instead of "issues" because I think it makes more sense to "solve a problem" than to "solve an issue". Also removed "may" from the second part.

```suggestion
- We aim to solve problems rapidly, prioritising them based on their impact.
```

 - We value and respect all differences in all people (seen and unseen). We aspire to inclusive working experiences and an environment that reflects the audience we serve, where our people take up career development opportunities equally, their voices are heard and contribute to our future. We aim to hire, retain and advance people from underrepresented groups.

**Working as a team**

We believe that the team is everything and our lowest common unit, set up with the necessary skills to succeed.

_Behaviours_:
 - We fail and succeed together as a team.
 - We are independent, accountable and share responsibility. We conduct ourselves with humility, kindness and a calm manner. We debate ideas, not people and we share constructive feedback openly. 
 - We take the time needed to listen to and help our colleagues with any problem they might have from anywhere within the business.
 - We value the experience of people in the team and the different constraints within their lives. 
 - We celebrate success and positive behaviour, recognizing everyone’s contribution while treating every moment including tasks and failures as a learning opportunity.

**We are data informed** 

> We believe that to be responsible in our impact we must be led by data.

Reviewer 3: ```suggestion
We believe that to be responsible in our impact we must be informed by data.
```

_Behaviours_: 
 - We base our decisions on analytics and experiments to reach consensus having heard everyone’s voice. 

>  - We measure our success through data analysis and not through the highest paid person's opinion or the loudest voice.

Reviewer 6: I'd like to propose a new item here, it's something I think is important, let me know if you agree?

> When we can't make an informed decision we're ready to say "we don't know". Then it's time to gather whatever new information we need.

Author: @maxspencer While I fully agree with the first part of the sentence, I think the second part is more nuanced. we sometime make decision even if we are uninformed or not enough because we have to move on. I did not came with a great wording so feel free to suggest one in a new PR

>  - We measure our success through data analysis and not through the highest paid person's opinion or the loudest voice.

Reviewer 7: This feels a bit disingenuous as it implies that data analysis is in some way objective. I suspect it also isn't true, the process of "data analysis" probably delivers something that allows a consensus to be generated about whether a change is successful.

Author: This is a fair point as objectivity is difficult(impossible) to achieve and one could always game a system to achieve what they would like. However I think this sentence is explicit enough on how we aim to measure success without defining a flawless protocol to do so.

**Empowerment**

We believe that all of us want to - and can - make a difference to the Guardian.

_Behaviours_: 
  - Meeting the needs of people we encounter across the Guardian is consistently encouraged across all levels of engineering.
  - We trust engineers to build the right things, collaborating closely and transparently with other teams.

>   - We own the problem, not the spec. We don’t wait for decisions to be made for us, we are an empowered engineering function and you are an empowered engineer. We use that power responsibly as we are conscious some decisions may need to be signed off and work collaboratively with our stakeholders.

Reviewer 7: This should probably be broken down into smaller points. Some of them are also beliefs rather than behaviours

Author: Any suggestion on how you would rewrite it?

**Technical Excellence** 

We believe that embracing technical innovation and excellence can help us make a difference.

_Behaviours_:
- We think we can better achieve the Guardian's objectives by having a deep understanding of the technology we use and adopting some emerging tools and solutions before others:

>     - We were one of the [first companies in UK to adopt `scala`](https://www.theguardian.com/info/developer-blog/2013/apr/26/how-the-guardian-built-scalable-foundation)

Reviewer 7: This is your history not your behaviour

Author: I have rewrote it to made clearer this history illustrate the behaviour

>     - We were one of the [first companies in UK to adopt `scala`](https://www.theguardian.com/info/developer-blog/2013/apr/26/how-the-guardian-built-scalable-foundation)
>     - We were one of the first companies [to use continuous delivery](https://www.theguardian.com/info/developer-blog/2015/jan/05/delivering-continuous-delivery-continuously)
>     - We were one of the [first legacy companies to move entirely to the cloud (AWS)](http://perspectives.mvdirona.com/2016/03/a-decade-of-innovation/)
>     - We were a pioneer of [the concept of microservices](https://martinfowler.com/articles/microservices.html#footnote-etymology)

Reviewer 8: Slightly more consistent:
```suggestion
    - We were one of the [first companies in UK to adopt `scala`](https://www.theguardian.com/info/developer-blog/2013/apr/26/how-the-guardian-built-scalable-foundation)
    - We were one of the [first companies to use continuous delivery](https://www.theguardian.com/info/developer-blog/2015/jan/05/delivering-continuous-delivery-continuously)
    - We were one of the [first legacy companies to move entirely to the cloud (AWS)](http://perspectives.mvdirona.com/2016/03/a-decade-of-innovation/)
    - We were a pioneer of the [concept of microservices](https://martinfowler.com/articles/microservices.html#footnote-etymology)
```

 - We maintain our technical advantage through continuous innovation

>  - We aim to be at the forefront of tech, staying curious, looking to innovate and continuously improve.

Reviewer 7: ```suggestion
 - We will be at the forefront of tech, always being curious, innovating and continuously improving.
```
This seems like an objective, I've suggested more directive language

**Knowledge sharing**

We believe that it’s our responsibility to upskill one another, sharing our knowledge, learnings and mistakes. 

_Behaviours_:
 - We practice pairing to help share our knowledge, grow our skills, and facilitate collaborative decision making.
 - We value pairing as a skill which we encourage people to grow, working together to find the styles which are most effective.
 - We make time for individual-learning & career development with personal development being our priority.
 - We write code for the most junior, not the most senior.
 - We actively share at all levels of users and stakeholders, whether it's our knowledge, work, planning or approach.

**We care**
 

> We choose to work here because we believe in [our values](https://workforus.theguardian.com/our-values) and what the Guardian stands for. We are not driven to simply deliver the implementation of a specification, but delight in working collaboratively to achieve an objective. In return, we are enthused to do whatever we can to deliver value to the Guardian.

Reviewer 6: To my mind, it would be better to shorten the first sentence:

```suggestion
We believe in [our values](https://workforus.theguardian.com/our-values) and what the Guardian stands for. We are not driven to simply deliver the implementation of a specification, but delight in working collaboratively to achieve an objective. We are enthused to do whatever we can to deliver value to the Guardian.
```

Whether our values are _the_ reason we choose to work here is neither here nor there. If they are then that's fine, but I think there are a lot of other good reasons (many of them mentioned above) that one might choose to work here.

_Behaviours_: 

>  - We seek to be a pioneer in the way things are done in the business and and aim to be seen as a model by other teams and departments

Reviewer 6: Nitpick: full stop

```suggestion
 - We seek to be a pioneer in the way things are done in the business and and aim to be seen as a model by other teams and departments.
```

>      

Reviewer 8: Is that our only caring behaviour?

Author: Feel free to suggest other ones with a new PR
